### PR TITLE
Feature/new proto

### PR DESCRIPTION
**Pull request by KaiL4eK**
New protocol for serial communication, tested showed good results, check pls

### CODE REVIEW

File: controller_wr_driver/include/link_def.h [hunk 1]
```diff
@@ -52,22 +48,22 @@ void ros_driver_send_pose( float x, float y, float dir, float vx, float uz );
...
+ * @brief	Send raw Steering value
  * @param
- * 			raw_adc - value from ADC [0; 4096] (12-bit)
+ * 			raw_adc - steering value [0; 4096] (12-bit)
  */
-void ros_driver_send_raw_adc( uint16_t raw_adc );
+void mproto_driver_send_raw_steering( uint16_t raw_steering );
 
 /***
  * WARNING! Never create this structure by yourself, use ros_driver_get_new_cb_ctx() instead
```
Comment (serykhelena) [inline controller_wr_driver/include/link_def.h]: Is this comment still relevant?

File: controller_wr_driver/include/link_def.h [hunk 2]
```diff
@@ -92,14 +88,14 @@ typedef struct
 	// control_params_setup_t (*get_control_params)( void );
 	// void (*set_control_params_cb)( control_params_setup_t * );
 
-} ros_driver_cb_ctx_t;
+} mproto_driver_cb_ctx_t;
 
 /**
  * @brief	Get callback context
  * @return
  * 			<ros_driver_cb_ctx_t> - structure with callback functions pointers
```
Comment (serykhelena) [inline controller_wr_driver/include/link_def.h]: and this one

File: controller_wr_driver/include/link_def.h [hunk 3]
```diff
@@ -109,10 +105,10 @@ ros_driver_cb_ctx_t ros_driver_get_new_cb_ctx( void );
  * 					To get default <ros_driver_cb_ctx_t> use ros_driver_get_new_cb_ctx()
```
Comment (serykhelena) [inline controller_wr_driver/include/link_def.h]: and here

File: controller_wr_driver/include/tests.h [hunk 4]
```diff
@@ -141,140 +135,12 @@ void testUARTControl( void );
...
-    testButtonRoutine( );
-
-#elif( MAIN_PROGRAM_ROUTINE == PROGRAM_ROUTINE_TEST_LIGHT )
-
-    testLightRoutine( );
-
-#elif( MAIN_PROGRAM_ROUTINE == PROGRAM_ROUTINE_TEST_LED_MATRIX )
-
-    testLedMatrixRoutine( );
+void testGUIRoutineServer ( void );
```
Comment (serykhelena) [inline controller_wr_driver/include/tests.h:146]: Maybe you can add short comments for these tests?

File: controller_wr_driver/include/uc_pc_link_defs.h [hunk 5]
```diff
@@ -0,0 +1,66 @@
+#ifndef UC_PC_LINK_DEFS_H_
+#define UC_PC_LINK_DEFS_H_
+
+/*** Input data ***/
+/*
+ * Velocity command data
+ * Data:
+ *  float[0] = velocity [mps]
+ *  float[1] = steering [deg]
```
Comment (serykhelena) [inline controller_wr_driver/include/uc_pc_link_defs.h:9]: Don't understand this comment. If we change value of define - we will get velocity?

File: controller_wr_driver/src/link_def.c [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+void reset_odometry_cb( void )
+{
+    if ( !last_cb_ctx.reset_odometry_cb )
+        return;
+
+    last_cb_ctx.reset_odometry_cb();
+}
+
+void cmd_vel_cb( float data[2] )
```
Comment (serykhelena) [inline controller_wr_driver/src/link_def.c:30]: float data[2] - from defines?

Comment (KaiL4eK) [reply controller_wr_driver/src/link_def.c:30]: Nope, that`s not connected

File: controller_wr_driver/include/uc_pc_link_defs.h (hunk 5 above)
Comment (KaiL4eK) [reply controller_wr_driver/include/uc_pc_link_defs.h:9]: We shouldn`t change the value - this file is prototype document for usage in all applications that are connected by one object .

**Pull request merged**